tests/rust/safe_abstraction/mutex_u32.rs

```diff
@@ -0,0 +1,444 @@
+#![feature(negative_impls)]
+#![allow(dead_code)]
+
+/*
+About the definitions:
+    Since in the proof of `MutexU32::lock` we need to close the `frac_borrow`,
+    in the `sys::locks::Mutex` interface; i.e. `SysMutex` fractions required by `sys::locks::Mutex::lock` to lock the mutex should be turened back from the method,
+    we cannot keep track of `SysMutex` fractions used to get the lock. So we adapth an interface for `sys::locks::Mutex` such that it does not care
+    if the mutex is locked or not.
+*/
```

Comment:
Please check for correct English spelling and grammar throughout this file. I have trouble parsing this sentence.